samples/net/mqtt/src/modules/network/network_lte.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+#include <modem/pdn.h>
+#include <modem/lte_lc.h>
+#include <nrf_modem.h>
+
+#include "message_channel.h"
+
+/* Register log module */
+LOG_MODULE_REGISTER(network, CONFIG_MQTT_SAMPLE_NETWORK_LOG_LEVEL);
+
+/* This module does not subscribe to any channels */
+
+/* Handler that is called when the modem hard faults. */
+void nrf_modem_fault_handler(struct nrf_modem_fault_info *fault_info)
+{
+	__ASSERT(false, "Modem error: 0x%x, PC: 0x%x", fault_info->reason,
+						       fault_info->program_counter);
+}
+
+/* Handler that is used to notify the application about LTE link specific events. */
+static void lte_event_handler(const struct lte_lc_evt *const evt)
+{
+	if ((evt->type == LTE_LC_EVT_MODEM_EVENT) &&
+	    (evt->modem_evt == LTE_LC_MODEM_EVT_RESET_LOOP)) {
+		LOG_WRN("The modem has detected a reset loop.");
+		LOG_WRN("LTE network attach is now "
+			"restricted for the next 30 minutes.");
+		LOG_WRN("Power-cycle the device to "
+			"circumvent this restriction.");
+		LOG_WRN("For more information see the nRF91 AT Commands - Command "
+			"Reference Guide v2.0 - chpt. 5.36");
+	}
+}
+
+/* Handler that notifies the application of events related to the default PDN context, CID 0. */
+void pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
+{
+	ARG_UNUSED(cid);
+
+	enum network_status status;
+
+	switch (event) {
+	case PDN_EVENT_CNEC_ESM:
+		LOG_DBG("Event: PDP context %d, %s", cid, pdn_esm_strerror(reason));
+		return;
+	case PDN_EVENT_ACTIVATED:
+		LOG_INF("PDN connection activated, IPv4 up");
+		status = NETWORK_CONNECTED;
+		break;
+	case PDN_EVENT_DEACTIVATED:
+		LOG_INF("PDN connection deactivated");
+		status = NETWORK_DISCONNECTED;
+		break;
+	case PDN_EVENT_IPV6_UP:
+		LOG_DBG("PDN_EVENT_IPV6_UP");
+		return;
+	case PDN_EVENT_IPV6_DOWN:
+		LOG_DBG("PDN_EVENT_IPV6_DOWN");
+		return;
+	default:
+		LOG_ERR("Unexpected PDN event!");
+		return;
+	}
+
+	__ASSERT_NO_MSG(zbus_chan_pub(&NETWORK_CHAN, &status, K_SECONDS(1)) == 0);
+}
+
+static void network_task(void)
+{
+	/* Setup a callback for the default PDP context. */
+	__ASSERT_NO_MSG(pdn_default_ctx_cb_reg(pdn_event_handler) == 0);
```

Comment:
All these `__ASSERT_NO_MSG` have to be removed.
They force the application to run with `__ASSERT` enabled, if that option is disabled nothing will work.

Reply:
Don't use `__ASSERT_NO_MSG` for error checking. All those functions that must be executed for this library to work, should be executed regardless of `__ASSERT`.

Reply:
Remember to not use `__ASSERT` for error checking; `__ASSERT` is not an error checking macro.
In general, you should check the result of the operations you execute (with an `if` condition). Otherwise, when `__ASSERT ` is disabled you won't have any idea whether the operation you tried to execute successful or not, and you won't be able to handle failures.

Reply:
Thanks for elaborating. Had to read up a bit on best practise use of assertions. I'll go over the sample and see if I can find reasonable spots to use them and where to not. Ill probably be replacing most of the instances with the usual `if` condition statement.

The idea was to give the sample an easier implementation with memfault which is able to capture coredumps when asserts are thrown. But, as you mention. Having them all over breaks with its intended use. User will have have to place these themselves I think in their debug scenario.